pkgs/development/ocaml-modules/terminal/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib, buildDunePackage, fetchurl, ocaml
+, stdlib-shims, uutf, uucp
+, alcotest, fmt
+}:
+
+buildDunePackage rec {
+  pname = "terminal";
+  version = "0.2.1";
+
+  minimalOCamlVersion = "4.03";
+  useDune2 = true;
+
+  src = fetchurl {
+    url = "https://github.com/CraigFe/progress/releases/download/${version}/terminal-${version}.tbz";
+    sha256 = "sha256:0vjqkvmpyi8kvmb4vrx3f0994rph8i9pvlrz1dyi126vlb2zbrvs";
+  };
+
+  propagatedBuildInputs = [ stdlib-shims uutf uucp ];
+
+  doCheck = lib.versionAtLeast ocaml.version "4.05";
+  checkInputs = [ alcotest fmt ];
```

Comment:
Also lists `astring` and `mtime` in opam file. Not sure if they are unneeded or propagated.

Reply:
The `opam` file is probably out of date: https://github.com/CraigFe/progress/blob/0.2.1/src/terminal/tests/dune

Reply:
Checks out.